services/QuillLMS/app/workers/identify_stripe_invoices_without_subscriptions_worker.rb

```diff
@@ -29,14 +29,35 @@ def perform
   private def relevant_stripe_invoices
     stripe_invoices.filter do |invoice|
       RELEVANT_INVOICE_STATUSES.include?(invoice.status) &&
-        !Subscription.exists?(stripe_invoice_id: invoice.id)
+        !invoice_ids_with_subscriptions.include?(invoice.id) &&
+        !invoice_refunded?(invoice) &&
+        non_zero_amount?(invoice)
     end
   end
 
+  private def invoice_ids_with_subscriptions
+    @invoice_ids_with_subscrirptions ||= Subscription.where(stripe_invoice_id: stripe_invoice_ids).pluck(:stripe_invoice_id)
```

Comment:
looks like you have a typo here

Reply:
It's important to cache your subscrirptions!